JQuery

yuriKIM-i3 opened the pull request:
### JQuery로 #6 처럼 만들기 ###

Commit:
JQuery

Commit:
날짜를 클릭하면 배경색이 빨간색으로 변함

crane93 commented:
자바스크립트때는 각 step마다 브랜치를 따로 만들었었죠,
jquery 브랜치의 목표는 step3를 만드는거니 
만약 자바스크립트 step1에 해당하는 과정이 끝나면 커밋하고 리뷰받을려고 합니다.

즉 step1,2이런식으로 브랜치를 안만들어보고 싶다는 뜻인데...
`센세의 생각은 어떻습니까`

Commit:
한번클릭하면 배경이 빨간색, 다시클릭하면 흰색

yuriKIM-i3 commented on use_JQuery/index.html:
```diff
@@ -5,7 +5,8 @@
   <title>Aug Calendar</title>
   <link href="style.css" rel="stylesheet" type="text/css">
-  <script src="scripts.js"></script>
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.0/jquery.min.js"></script>
```
jquery를 쓰는 방법이 두가지가 있다고 하는데,
* jquery홈페이지에서  jquery파일을 다운받기
* 인터넷에서 CDN을 검색해서 쓰기
사회인들은 어떤방법을 많이 쓰나요? 🙄

cheekykorkind replied:
경험상 두가지중에서 고르는듯 합니다.
- 간단한걸 선호하는 사람은 CDN
- 페이지 로드되는 성능도 생각할 필요가 있을때는 다운로드

yuriKIM-i3 commented on use_JQuery/index.html:
```diff
@@ -66,5 +69,21 @@ <h1>8월달력</h1>
+      $(this).addClass('setRed');
+      }
+  });
+</script>
```
script의 위치를 처음에 head에다가 썼더니 안되더라구요. 
검색해보니 자바스크립트처럼 html이 먼저 로드되야 jquery된다고 하네요.
그럼 제가 적은 위치는 right한겁니까 인간? 🙉

cheekykorkind replied:
그렇다고 생각합니다!

yuriKIM-i3 replied:
자바스크립트같이 따로 파일 만들어서 작성 안하나여?

cheekykorkind replied:
따로 만들어서 작성 부탁드립니다!

Commit:
주말을 클릭하면 알람 생성

Commit:
날짜랭킹 추가

cheekykorkind commented on use_JQuery/jquery.js:
```diff
@@ -0,0 +1,85 @@
+// 그럼 로드를 기다려주는 코드를 적어보고 해보겠습니다.
+
+$( document ).ready(function() {
+    ChangBackColor();
```
일반적으로 명명규칙은 Camel case나  Snake case둘중에 하나를 많이 씁니다. 본 프로젝트는 Camel case를 쓰고 있으니, 변수명, 함수명은 소문자로 시작합시다! ( `https://en.wikipedia.org/wiki/Naming_convention_(programming)` )

cheekykorkind commented on use_JQuery/jquery.js:
```diff
@@ -0,0 +1,85 @@
+$( document ).ready(function() {
+    ChangBackColor();
+    alertWeekends();
+    EasterEgg();
```
Camel case는 변수명, 함수명은 소문자로 시작합시다!

cheekykorkind commented on use_JQuery/jquery.js:
```diff
@@ -0,0 +1,85 @@
+            previouSetRedElement.removeClass('setRed');
+            $(this).addClass('setRed');
+        }
+        CountClicked(getDay, clickedcount);
```
Camel case는 변수명, 함수명은 소문자로 시작합시다!

cheekykorkind commented on use_JQuery/index.html:
```diff
@@ -5,7 +5,8 @@
   <title>Aug Calendar</title>
   <link href="style.css" rel="stylesheet" type="text/css">
-  <script src="scripts.js"></script>
+  <script src="http://code.jquery.com/jquery-1.11.0.min.js"></script>
```
May 24, 2013쯤 버전이니 6년전이군요.
JQuery 버전이 낮을수록 버그가 남아있을 가능성이 높으므로 되도록 최신버전을 씁시다!
https://en.wikipedia.org/wiki/JQuery

cheekykorkind commented on use_JQuery/jquery.js:
```diff
@@ -0,0 +1,85 @@
+// 현재 1순위는ㄴ HTML 로드를 기다리는 코드가 없어서 그런거 같습니다.
```
필요한 주석입니까?

cheekykorkind commented on use_JQuery/jquery.js:
```diff
@@ -0,0 +1,85 @@
+  clickedcount[i]=0;
+}
+
+function ChangBackColor(){
```
함수에 주석을 다는 습관을 강력히 추천합니다.
```
/**
 * 함수가 달성하고자하는 목적을 적습니다.
 *
 * (함수 파라미터가 없으면 안적습니다.)@param 타입 변수명 필요하다면 간단한 설명
 * 예시) @param string var textNodeContent 텍스트 노드의 텍스트부분 입니다.
 * 
 * (함수 리턴이 없으면 안적습니다.)@return 타입 변수명 필요하다면 간단한 설명
 * 예시) @return int var arrayCount 배열의 갯수입니다.
 */
```

cheekykorkind commented on use_JQuery/jquery.js:
```diff
@@ -0,0 +1,85 @@
+});
+
+var clickedcount = new Object();
+for(var i=1;i<=31;i++){
```
연산자(=, +등)와 변수명, 값의 띄어쓰기는 통일합시다. 붙여쓰기를 택한다면 모든 코드에서 붙여쓰고, 한칸 띄운다면 모든 코드에서 한칸 띄워서 일관성을 유지하면 좋을듯합니다.

cheekykorkind commented on use_JQuery/jquery.js:
```diff
@@ -0,0 +1,85 @@
+        return;
+    }
+    clickedcount[getDay]++;
+    var r = `클릭한 날짜 ${getDay}일은 ${clickedcount[getDay]}번 눌렀습니다.`;
```
변수명은 되도록 의미를 표현할 수 있게 적읍시다!

cheekykorkind commented on use_JQuery/jquery.js:
```diff
@@ -0,0 +1,85 @@
+}
+
+function previousLine(getDay, clickedcount){
+    console.log($("#day"+getDay+"").length);
```
디버그용 코드는 지웁시다!

cheekykorkind commented on use_JQuery/jquery.js:
```diff
@@ -0,0 +1,85 @@
+    }
+
+    clickedcount[getDay]++;
+    $("#day"+getDay+"").html(`클릭한 날짜 ${getDay}일은 ${clickedcount[getDay]}번 눌렀습니다.`);
```
`"#day"+getDay` 로 충분할듯 합니다. 확인 부탁드립니다!

cheekykorkind reviewed (changes requested):
확인 부탁드립니다!

cheekykorkind commented:
> 名づけ能力が必要だと書いていますが…
その力はどうすれば手に入れるっすか

二つありまして、
一つ目は自分でコードをたくさん書きながらたくさん名前を考えること。
二つ目は他人のコードを読んで名前をちゃんと見ることです。Githubの有名なプロジェクト推奨します。( `https://github.com/jquery/jquery` など)

Commit:
리뷰 반영

cheekykorkind commented on use_JQuery/jquery.js:
```diff
@@ -1,21 +1,23 @@
+    easterEgg();
 });
 
 var clickedcount = new Object();
```
코드상에 전부 clickedCount로 바꾸고 머지 부탁드립니다!

cheekykorkind reviewed (approved):
변수명을 clickedCount로 바꾸고 머지 부탁드립니다!

Commit:
두번째 수정사항 반영

yuriKIM-i3 commented on use_JQuery/jquery.js:
```diff
@@ -26,51 +26,51 @@ function changBackColor(){
-    clickedcount[clickedDay]++;
-    var showHowManyClicked = `클릭한 날짜 ${clickedDay}일은 ${clickedcount[clickedDay]}번 눌렀습니다.`;
+    clickedCount[clickedDay]++;
+    var showHowManyClicked = `클릭한 날짜 ${clickedDay}일은 ${clickedcount[clickedCount]}번 눌렀습니다.`;
```
> 변수명은 되도록 의미를 표현할 수 있게 적읍시다!

라고 해서 이렇게 바꿔봤습니다
좀 너무 긴것 같은데... 비록 길지만 의미를 담는게 좋은가요?
`showHowManyClicked라는 이름은 어떤가요?`

cheekykorkind replied:
- 변수명이 길더라도 코드의 의미를 아는쪽을 선호하기 떄문에 좋다고 생각합니다.
- 변수명도 함축적이고 주석도 없으면 읽는쪽(다른사람, 미래의 자신)이 편하지 않기때문에 변수명 긴쪽을 선호합니다.

crane93 commented:
혹시 
https://github.com/unnnyong/Charkak_Cheese/commit/f140cf551238d62e43ecb1fd3f3b86c2ba6f66d8
처럼 커밋에 메세지 쓰는법 아시나요오 🏄‍♀

cheekykorkind commented:
`git commit -m "쓰고 싶은 메세지"` 로 메세지 적으면 됩니다!

Commit:
jQeury쓰며 느낀점

1. jQuery도 자바스크립트처럼 html이 로드된 뒤 실행되어야함
jQuery파일에 .ready(fuctin(){});를 맨 첫출에 써주고 이 안에 함수를 적자

2.(20번째줄) .text()로 가져온 값을 정수로 변환
하지만 공백을 가져왔을 땐 NaN라고 숫자가 아니라고 뜸
|| 0 으로 NaN은 0으로 처리하자

3.(61번째줄)#dicday가 존재하는지 검사가 필요했음
().length===0으로 기존에 divday가 있는지 찾을 수 있었다
